development_environments/using_selenium_with_vagrant.md

```diff
@@ -0,0 +1,83 @@
+## [Home](../README.md)
+
+# Using Selenium with Vagrant
+The Selenium web driver for Capybara makes it easy to write integration tests because you can see and interact with the web browser that is running your tests. A downside of Selenium is it runs slower than headless drivers like PhantomJS. So a common approach is to use Selenium while you're writing or debugging tests, and then switch to a faster headless driver after the test is working.
+
+Vagrant works great for setting up isolated development environments, but since Vagrant runs in headless mode, it doesn't work well with Selenium.
+
+Here's how to make them play nicely together.
+
+## Setup
+
+### 1. Install Dependencies on Mac
+```
+$ brew install chromedriver
+$ brew install selenium-server-standalone
+```
+
+### 2. Configure Capybara and VCR
+Add this code to your test suite's config file (e.g. `features/env.rb`).
+
+```ruby
+SELENIUM_SERVER = '10.0.2.2' # found with `netstat -rn | grep "^0.0.0.0 " | cut -d " " -f10`
+SELENIUM_PORT = 4444
+RAILS_APP_HOST = '127.0.0.1:3001'
```

Comment:
Instead of having set and needed to run could do something like this

```
Capybara.configure do |config|
  config.default_driver = :poltergeist
  config.run_server = true
  config.server_port = 31338
end
```

That will ensure test server is always booted on 31338.  Which I think maybe a default for Capybara.